### DISCUSSION TITLE
Request for statistic role: Reporter

### DISCUSSION THREAD

**Discussion by khacminh** [community/community#38263] [Repositories]
Hello,

I would like to request a new static repo role: **Reporter**. This user:

- Can create/Read issues
- Can comment on issues
- **Cannot** access the source code

I believe this role is widely needed for the organizations in GitHub

**Comment by jfullstackdev** [comment]
why do you believe this role would be great ? 

anyone in the team should have the access to the code, GitHub was built on that idea, there is no problem implementing it but if it's not really that needed, then I doubt whether that will be implemented

if your role will be triaging issues, then go ahead, you can apply for that in any open source and if even the `Reporter` role will be implemented, it will still be the approval of team so that one can join the team to be a `Reporter`, same effort done

**Reply by khacminh** [reply]
Hi @jdevstatic,

Thank you for your reply.

In my opinion, your idea is from the developer's view. However, in a project, there are many roles and not all of them should have permission to access the source code. You're right about the open-source repos. But GitHub also provides Team and Enterprise subscriptions. And for organizations, I believe a `Reporter` user is a better approach than an empty repo called [issues-only repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-an-issues-only-repository).

**Reply by jfullstackdev** [reply]
yup, in a project, there is the Project Manager, the Tech Lead, the Managing Director etc., plus those if they are implementing Agile Sprint or DevOps

in my company, mostly project managers will not have access to the repo even if they were given access, barely they will look at it, because it's not simply their focus but rather on the project being done, whether the code implementation is correct, the one that knows much of the technicalities is the Tech Lead and the Developers

even if say, in a group chat, they are chatting with all the technicalities, the Projects Managers will simply look at it, to catch up what's going on,

as for your requested `Reporter`, as I've mentioned, it's the same effort that will be done by the team, they will still be giving access to that user for them to, say, triage the issue, it does not make any great difference whether the reporter is seeing or not the repos because the focus of them are the issues

and again, I'm not opposing the request here, I'm just mentioning that GitHub also have to decide of all the features they receive, just like this Discussion feature, before this, I also emailed them requesting a Discussion tab that is different from Issues because Issues tab carriers the negative word instead of the term Discussion

**Reply by khacminh** [reply]
Hi @jdevstatic

Thank you for your reply.

In your example, there is someone that doesn't need to access the source code. And of course, there are other roles in the project that don't need or even should not access the source code. And if everyone can clone the source code, it makes the job of preventing the source code from being leaked or stolen more and more difficult.

That's why I suggest a new role having the similar permissions as Read but cannot access the source code.
